target_chains/near/receiver/tests/workspaces.rs

```diff
@@ -940,3 +967,164 @@ async fn test_governance_target_fails_if_not_near() {
+            .map(|f| to_vec::<_, BigEndian>(f).unwrap())
+            .collect();
+        let all_feeds_bytes_refs: Vec<_> = all_feeds_bytes.iter().map(|f| f.as_ref()).collect();
+        let tree = MerkleTree::<Keccak160>::new(all_feeds_bytes_refs.as_slice()).unwrap();
+        let mut price_updates: Vec<MerklePriceUpdate> = vec![];
+        for update in updates {
+            let proof = tree
+                .prove(&to_vec::<_, BigEndian>(update).unwrap())
+                .unwrap();
+            price_updates.push(MerklePriceUpdate {
+                message: PrefixedVec::from(to_vec::<_, BigEndian>(update).unwrap()),
+                proof,
+            });
+        }
+        create_accumulator_message_from_updates(
+            price_updates,
+            tree,
+            corrupt_wormhole_message,
+            [1; 32],
+            wormhole::Chain::Any.into(),
+        )
+    }
+
+    let (_, contract, _) = initialize_chain().await;
+
+    // Submit a new Source to the contract, this will trigger a cross-contract call to wormhole
+    let vaa = wormhole::Vaa {
+        emitter_chain: wormhole::Chain::Any,
+        emitter_address: wormhole::Address([0; 32]),
+        sequence: 1,
+        payload: (),
+        ..Default::default()
+    };
+
+    let vaa = {
+        let mut cur = Cursor::new(Vec::new());
+        serde_wormhole::to_writer(&mut cur, &vaa).expect("Failed to serialize VAA");
+        cur.write_all(
+            &GovernanceInstruction {
+                target: Chain::from(WormholeChain::Any),
+                module: GovernanceModule::Target,
+                action: GovernanceAction::SetDataSources {
+                    data_sources: vec![
+                        Source::default(),
+                        Source {
+                            emitter: [1; 32],
+                            chain:   Chain::from(WormholeChain::Any),
+                        },
+                    ],
+                },
+            }
+            .serialize()
+            .unwrap(),
+        )
+        .expect("Failed to write Payload");
+        hex::encode(cur.into_inner())
+    };
+
+    assert!(contract
+        .call("execute_governance_instruction")
+        .gas(300_000_000_000_000)
+        .deposit(300_000_000_000_000_000_000_000)
+        .args_json(&json!({
+            "vaa": vaa,
+        }))
+        .transact_async()
+        .await
+        .expect("Failed to submit VAA")
+        .await
+        .unwrap()
+        .failures()
+        .is_empty());
+
+    // Create a couple of test feeds.
+    let feed_1 = create_dummy_price_feed_message(100);
+    let feed_2 = create_dummy_price_feed_message(200);
+    let message = create_accumulator_message(&[feed_1, feed_2], &[feed_1], false);
+    let message = hex::encode(message);
+
+    // Call the usual UpdatePriceFeed function.
+    assert!(contract
+        .call("update_price_feed")
+        .gas(300_000_000_000_000)
+        .deposit(300_000_000_000_000_000_000_000)
```

Comment:
lol near 

we're going to have to help people generate these values in the API I think